### PR TITLE
Fixed offspring bugs, added to expansiontravel.gd, unique npc bug, etc.

### PR DESCRIPTION
**Pull request by Ralphomayo**
Fixed offspring bodyparts and race not being assigned as intended and related bugs. Tweaked ralphstweaks version of body part assignments. Streamlined constructor.gd as relates to offspring generation. Added more to expansiontravel.gd (bug fixes and new content from community). Fixed unique npc >100% bug when playing base Aric's mod (ralphstweaks disabled).

### CODE REVIEW

File: customScripts/expansiontravel.gd [hunk 1]
```diff
@@ -2,17 +2,21 @@
...
 func getzonetraveltext(zone,progress):
 	var array = ["Your journey continues peacefully. "]
 	var player = globals.player
 	var party = globals.state.playergroup.duplicate()
 	var teammates = []
-	for i in party:
-		var j = globals.state.findslave(i)
-		if globals.player != j:
-			teammates.append(j)
+	if party.size() > 0:
```
Comment (ankmairdor) [inline customScripts/expansiontravel.gd]: This line adds nothing, as for loops do not iterate empty containers nor do they produce errors

File: customScripts/expansiontravel.gd [hunk 2]
```diff
@@ -118,9 +126,9 @@ func getzonetraveltext(zone,progress):
 		elif captives.size() > 5 && globals.state.reputation.wimborn > 50 && captivebaddiesratio > 69:
 			array.append("A large "+travelgroup()+" of " +travelcheerful()+" "+randomfromarray(['townsfolk','revelers','students','tradesmen','guards'])+" returning to town cheers you and your party along after they see your "+travelcaptivedesc()+" captives shuffling along in their restraints.")
 		if rand_range(0,1) > 0.9 && globals.expansionsettings.sillymode:
-			array.append("You notice a "+travelcheerful()+" "+travelrandomperson()+" "+travelfucking()+" "+travelrandomadj()+" "+travelfucktarget()+" - in broad daylight.\n\n"+travellink()+"\n\nIt's clearly a "+travelreaction()+" moment for him and a "+travelreaction()+" moment for you.")
+			array.append_array(["You notice a "+travelcheerful()+" "+travelrandomperson()+" "+travelfucking()+" "+travelrandomadj()+" "+travelfucktarget()+" - "+travelobserved()+".\n\n"+travellink()+"\n\nIt's clearly a "+travelreaction()+" moment for him and a "+travelreaction()+" moment for you."])
```
Comment (ankmairdor) [inline customScripts/expansiontravel.gd]: This does not need to use append_array as there is only a single string

File: scripts/characters/constructor.gd [hunk 3]
```diff
@@ -701,20 +727,30 @@ func setRace(person,mother):
...
-			if primaryracepercent > secondaryracepercent:
+			if person.genealogy[race] > primaryracepercent:
 				secondaryrace = primaryrace
 				secondaryracepercent = primaryracepercent
+				primaryrace = race
+				primaryracepercent = person.genealogy[race]
+				racetiestobreak = []
+			#ralphD
+			elif person.genealogy[race] == primaryracepercent:
+				racetiestobreak.append_array([race])
```
Comment (ankmairdor) [inline scripts/characters/constructor.gd]: there is no reason to use append_array() over append() here, it's just inefficient. also for the line below this

File: customScripts/expansiontravel.gd [hunk 4]
```diff
@@ -2,17 +2,21 @@
 extends Node
 
 func randomfromarray(array):
-	return array[randi() % array.size()]
+	if array.size() > 0:
+		return array[randi() % array.size()]
+	else:
+		return []
```
Comment (ankmairdor) [inline customScripts/expansiontravel.gd]: Why return an array? This makes no sense to me. Returning null at least fits standard expectations, but this function  shouldn't even be called for empty arrays.

Comment (Ralphomayo) [reply customScripts/expansiontravel.gd]: My bug fixing involves launching spaghetti at a wall apparently.  It did work, but may have made a mess in the process.  -_-

File: customScripts/expansiontravel.gd [hunk 5]
```diff
@@ -48,7 +52,11 @@ func getzonetraveltext(zone,progress):
 		#if !globals.state.backpack.stackables.has('bandage') && i.health < i.max_health/2:
 		#	text =
 		#	break
-	if rand_range(0,1) > 0.5:
+	if rand_range(0,1) > 0.9:
+		array.append_array(["A flock of crows pick at a well scavenged corpse nearby.  Upon examination of the nearly stripped bones and tattered remains of clothing, you figure this poor traveler was likely a poor sword for hire or vagabond.  It's unclear what killed him.  Whatever possessions he might have once had have long been looted or dragged away by vermin.  Unable to determine even their place of origin or likely destination, you decide to move on."])
```
Comment (ankmairdor) [inline customScripts/expansiontravel.gd]: Both of these entries do not need append_array() as they contain only a single string

File: customScripts/expansiontravel.gd [hunk 6]
```diff
@@ -177,6 +185,8 @@ func getzonetraveltext(zone,progress):
...
 		else:
 			array.append("There's writing on the wall here.  "+randomfromarray(["It's some dialect of elvish, but you cannot make it out","It's illegible to you, but an arrow pointing onward with an 'X' carved through it and skulls convey the meaning well enough."]))
+	elif zone.code == 'undercityruins':
+		array.append_array(["Your party turns a corner and comes face to face with a wall.  You turn around to backtrack and dozens of sightless eyes glisten in the darkness from alcoves you couldn't see on your way into this dead end.\n\nYou prepare for the end as one massive trogolodyte moves to the center of the tunnel ahead of the others, but before it can act a Kobold wearing some sort of jersey drops from the ceiling and bravely slaps the monstrosity on its nose as it falls before scurrying through the trog's legs and behind it.\n\nThe Kobold races down the hall screaming it's warcry, [color=yellow]Got your nose![/color].  The enraged trogolodytes forget all about your party and give chase, leaving you alone in relative safety.\n\nYou're relatively certain that the Kobold's jersey read: 'Capitualize'."])
```
Comment (ankmairdor) [inline customScripts/expansiontravel.gd]: This entry does not need append_array() as it is a single string

File: scripts/characters/constructor.gd (hunk 3 above)
Comment (Ralphomayo) [reply scripts/characters/constructor.gd]: claw hammer?  well that is a lot better than the sledge I found somewhere else, I'll give that a try.  ;]

Comment (Ralphomayo) [reply scripts/characters/constructor.gd]: comment incorporated in new commit

File: customScripts/expansiontravel.gd (hunk 1 above)
Comment (Ralphomayo) [reply customScripts/expansiontravel.gd]: comment incorporated for new commit

File: customScripts/expansiontravel.gd (hunk 4 above)
Comment (Ralphomayo) [reply customScripts/expansiontravel.gd]: comment incorporated on new commit

File: customScripts/expansiontravel.gd (hunk 5 above)
Comment (Ralphomayo) [reply customScripts/expansiontravel.gd]: comment incorporated on new commit

File: customScripts/expansiontravel.gd (hunk 6 above)
Comment (Ralphomayo) [reply customScripts/expansiontravel.gd]: comment incorporated on new commit

File: customScripts/expansiontravel.gd (hunk 2 above)
Comment (Ralphomayo) [reply customScripts/expansiontravel.gd]: comment incorporated on new commit

**Pull request merged**